sanic/app.py

```diff
@@ -1453,9 +1452,34 @@ async def __call__(self, scope, receive, send):
     # -------------------------------------------------------------------- #
     # Configuration
     # -------------------------------------------------------------------- #
+
     def update_config(self, config: Union[bytes, str, dict, Any]):
         """Update app.config.
 
         Please refer to config.py::Config.update_config for documentation."""
 
         self.config.update_config(config)
+
+    # -------------------------------------------------------------------- #
+    # Class methods
+    # -------------------------------------------------------------------- #
+
+    @classmethod
+    def register_app(cls, app: "Sanic") -> None:
```

Comment:
These methods would be great candidates to the new [`final`](https://docs.python.org/3/library/typing.html#typing.final) typing hint, but that's only available for Python 3.8+ ... I think we should, at least, provide some kind of warning in the docs to avoid overriding these methods - unless you're **absolute sure** of what you're doing. This way we won't need to even _enforce_ this being `final` (I don't like enforcing things in Python) ...

Reply:
There's a long list if we want to go down that route. I think this is not the place for that.

Reply:
Let's circle back to the typehint later, I think this is fine for now.